### PR TITLE
Spire workload registrar

### LINKED ISSUE #13: Spire: update chart to use k8s-workload-registrar as opposed to init container
https://github.com/spiffe/spire/tree/main/support/k8s/k8s-workload-registrar

### PR DESCRIPTION
**Pull request by marcofranssen**
This PR replaces the manual init containers with the `k8s-workload-registrar`.

Workflow to test

```bash
kubectl delete ns test-spire
kubectl create ns test-spire
helm -n test-spire install awesome charts/spire
# verify spire entries
kubectl exec -n test-spire awesome-spire-server-0 -c spire-server -- \
    bin/spire-server entry show -registrationUDSPath /run/spire/sockets/registration.sock
helm -n test-spire install stuff charts/spire-client-example
# verify spire entries
kubectl exec -n test-spire awesome-spire-server-0 -c spire-server -- \
    bin/spire-server entry show -registrationUDSPath /run/spire/sockets/registration.sock
```

WIP aligning with these configs:

https://github.com/spiffe/spire/tree/main/test/integration/suites/k8s-reconcile/conf

resolves #13 

In general there are still a few edge cases to support in the chart.

- [x] **docker-for-mac** has an issue in the `spire-agent` due to _serviceAccountTokenProjections volume_
     - We might want to add a config flag to deploy the chart with an alternative to the  _serviceAccountTokenProjections volume_
     - [x] Has SPIFFE entries available for the docker-for-mac node, spiffe server and k8s-registrar
- [ ] **Kind** has an issue in the `spire-agent` due to serviceAccountTokenProjections volume 
    - Trying to resolve that in following branch (https://github.com/philips-labs/helm-charts/compare/spire-workload-registrar...ci-kind-config) by applying Kind config as shown here https://github.com/spiffe/spire/tree/main/test/integration/suites/k8s-reconcile/conf , however this still fails https://github.com/philips-labs/helm-charts/runs/2896727241?check_suite_focus=true#step:7:714
- [x] **EKS** deployment succeeds, but don't see any SPIFFE entries in the server

    ```bash
    $ bin/spire-server entry show -registrationUDSPath /run/spire/sockets/registration.sock
    Found 0 entries
    ``` 

    **workload-registrar** (log):

    ```bash
    E0623 15:39:25.309569 25 leaderelection.go:324] error initially creating leader election record: configmaps is forbidden: User "system:serviceaccount:spire:spire-server" cannot create resource "configmaps" in API group "" in the namespace "spire": RBAC: clusterrole.rbac.authorization.k8s.io "spire-server-role" not found
    ```

In summary:

- _serviceAccountTokenProjections volume_ doesn't work everywhere natively.
- EKS deploys successfully, but doesn't show any SPIFFE entries in the server.

### CODE REVIEW

**Review by gjkamstra — COMMENTED** [review]
No comments at this moment.

**Comment by marcofranssen** [comment]
Might need to add "create" to the list of verbs in this block. https://github.com/philips-labs/helm-charts/blob/spire-workload-registrar/charts/spire/templates/server-role.yaml#L22-L25 to resolve k8s error.

also docker for mac has rbac disabled by default, so if you enable rbac youll probably see the same error: https://github.com/docker/for-mac/issues/3694#issuecomment-619474504

**Comment by marcofranssen** [comment]
https://github.com/docker/for-mac/issues/5784 this issue describes a bug in the current Docker-for-mac k8s setup and how to fix it.

**Comment by marcofranssen** [comment]
https://github.com/kubernetes-sigs/kind/issues/2327

**Comment by marcofranssen** [comment]
https://github.com/kubernetes-sigs/kind/issues/2327#issuecomment-867926334

File: charts/spire/README.md [hunk 1]
```diff
@@ -2,10 +2,31 @@
...
+```
+
+Then add the following to `/etc/kubernetes/manifests/kube-apiserver.yaml`
+
+```yaml
+spec:
+containers:
+- command:
+  - kube-apiserver
+  - --service-account-api-audiences=api
```
Comment (gjkamstra) [inline charts/spire/README.md:25]: Do we need to add api-server here?

File: charts/spire/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+  ```
+
+  Then add the following to `/etc/kubernetes/manifests/kube-apiserver.yaml`
+
+  ```yaml
+  spec:
+  containers:
+  - command:
+    - kube-apiserver
+    - --service-account-api-audiences=api
```
Comment (gjkamstra) [inline charts/spire/Chart.yaml:22]: Add api-server

**Review by gjkamstra — COMMENTED** [review]
One comments on missing api-server in docs

**Review by gjkamstra — APPROVED** [review]
Looks good
